client/src/components/Conversation/ConversationChat/ConversationChat.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react';
+import useStyles from './useStyles';
+import MessageInput from '../MessageInput/MessageInput';
+import { Box, Container, Typography } from '@material-ui/core';
+import { User } from '../../../interface/User';
+import getMessages from '../../../helpers/APICalls/getMessages';
+import sendAndSavaMessage from '../../../helpers/APICalls/sendMessage';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSocket } from '../../../context/useSocketContext';
+import { useConversation } from '../../../context/useConversationContext';
+import moment from 'moment';
+import './chat.css';
+
+const ConversationChat = (): JSX.Element => {
+  const classes = useStyles();
+  const { socket } = useSocket();
+
+  const { currentConversation, conversations, updateConversationContext } = useConversation();
+  const { loggedInUser } = useAuth();
+  const [sender, setSender] = useState<User | undefined>();
+  const [reciever, setReciever] = useState<User | undefined>();
+  const [messages, setMessages] = React.useState<
+    Array<{
+      senderId?: string;
+      text?: string;
+      sentTime?: string;
+    }>
+  >([]);
+
+  useEffect(() => {
+    if (!currentConversation) return;
+    if (currentConversation && loggedInUser && loggedInUser._id === currentConversation.senderId._id) {
```

Comment:
Why `senderId._id`?
You can also shorten this line with ES6 `optional chaining`, just nice to have:
`if (loggedInUser?._id === currentConversation?.senderId) ` OR
`if (loggedInUser?._id === currentConversation?.senderId?._id) `, if `senderId._id` is the correct one.

Reply:
Hi, this one is to check the current `user` is the sender or not, so I check the `senderId._id`